wallet/src/de/schildbach/wallet/ui/EncryptNewKeyChainDialogFragment.java

```diff
@@ -109,7 +114,10 @@ protected void onBadPassword() {
                 unlockButton.setEnabled(true);
                 unlockButton.setText(getText(R.string.wallet_lock_unlock));
                 pinView.setEnabled(true);
-                pinRetryController.failedAttempt(password);
+                if(pinRetryController.failedAttempt(password)) {
+                    restartService.performRestart(getActivity(), true);
+                    dismiss();
+                }
```

Comment:
In theory this dialog is not used anymore due to the changes with to resolve the Non-encrypted Wallets crashes and errors.

We can get rid of it in a new story.

Reply:
Nevermind, this one is still used in very rare cases.